Assets/NPCEngine/Demo/BasicDemo/Scripts/ButtonDisableEnable.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections;
+using System.Linq;
+using NPCEngine.Components;
+using NPCEngine.Utility;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ButtonDisableEnable : MonoBehaviour
+{
+    private Button _button;
+    public string serviceName;
+    void Start()
+    {
+        _button = gameObject.GetComponent<Button>();
+        _button.interactable = false;
+        
+        CoroutineUtility.StartCoroutine(EnableButtonWhenServiceRunning(), 
+            this, "EnableButtonWhenServiceRunning");
+    }
+
+    public IEnumerator EnableButtonWhenServiceRunning()
+    {
+        while (true)
+        {
+            
+            var services = NPCEngineManager.Instance.Services;
+            if (services != null && services.Count(serviceMetadata => serviceMetadata.service == serviceName) > 0)
```

Comment:
You should check NPCEngineManager.Instance.ServiceStatuses here as well to be running (the ordering should be the same as in NPCEngineManager.Instance.Services)